cells/lib/ops/mkStandardOCI.nix

```diff
@@ -68,20 +68,82 @@ in
       then debugOperable
       else operable;
 
-    setupLinks = cell.ops.mkSetup "links" [] ''
-      mkdir -p $out/bin
-      ${runtimeEntryLink}
-      ${debugEntryLink}
-      ${livenessLink}
-      ${readinessLink}
+    setupLinks =
+      cell.ops.mkSetup "links" [
+        {
+          regex = "/bin";
+          mode = "0555";
+        }
+      ] ''
+        mkdir -p $out/bin
+        ${runtimeEntryLink}
+        ${debugEntryLink}
+        ${livenessLink}
+        ${readinessLink}
+      '';
+
+    setupCerts = cell.ops.mkSetup "ssl" [] ''
+      mkdir -p $out/etc
+      cp -r ${nixpkgs.cacert}/* $out
+    '';
+
+    setupEtcHack = cell.ops.mkSetup "etc" [] ''
+      mkdir -p $out/etc
+      cat << REASON > $out/etc/.ensure-etc-is-no-symlink
+        # docker creates some minimal files in /etc at runtime
+        # that would otherwise shadow a top-level symlink:
+        #   bash-5.2$ ls /etc
+        #   hostname  hosts  mtab  resolv.conf
+        # therefore, we make trick nixpks.buildenv into not
+        # generating symlinks at the level of /etc but at least
+        # one level deeper by artifie of this overlapping element
+        # that buildenv will detect and push the linking one level down
+      REASON
+    '';
```

Comment:
@gytis-ivaskevicius

Reply:
but also @nlewo

Reply:
I am not sure how this changes anything. Is it a special exception in `buildEnv` function?

Reply:
`buildEnv` has a shallow-first linking strategy, when we inhibit top level linking by this fake entropy, `buildEnv` will link one level down, instead.